src/cli/CliHandler.cc

```diff
@@ -82,149 +83,148 @@ void CliHandler::logo()
+void CliHandler::parse_arguments()
+{
+    app = new CLI::App("DRAMPower");
+
+    try {
+        app->add_flag(IO_TERM,
+                      io_term_active,
+                      "IO and Termination");
+
+        app->add_flag(DEBUG_FILE,
+                      debug_file_active,
+                      "Generate debug file");
+
+        app->add_flag(DEBUG_CONSOLE,
+                      debug_console_active,
+                      "Display debug messages on Console");
+
+        app->add_flag_function(VERS,
+                               [&](bool){
+            logo();
+            std::exit(EXIT_SUCCESS);
+        },
+        "Display DRAMPower version information");
+
+        app->add_option(MEM_SPEC,
+                        mem_spec_path,
+                        "Memory specification file")
+                ->required()
+                ->check(CLI::ExistingFile);
+        app->add_option(CMD_TRACE,
+                        cmd_trace_path,
+                        "Commands trace file")
+                ->required()
+                ->check(CLI::ExistingFile);
+
+        app->parse(argc, argv);
+
+    } catch (const CLI::ParseError &e) {
+        app->exit(e);
+        std::exit(EXIT_FAILURE);
+    } catch (const std::exception &e) {
+        throw e;
+    } catch (...) {
+        throw std::runtime_error("Unknown error during arguments parsing!");
+    }
 }
 
-bool CliHandler::get_io_term_active() const{
-  return io_term_active;
+bool CliHandler::get_writeToConsole() const
+{
+    return debug_console_active;
 }
 
-const std::string& CliHandler::get_mem_spec_path() const{
-  return mem_spec_path;
-}
 
-const std::string& CliHandler::get_cmd_trace_path() const{
-  return cmd_trace_path;
+bool CliHandler::get_writeToFile() const
+{
+    return debug_file_active;
 }
 
-bool CliHandler::get_bank_wise_active() const{
-  return bank_wise_active;
-}
 
-int CliHandler::get_bank_wise_rho() const{
-  try{
-    return bank_wise_parms.at(0);
-  } catch (const std::exception& e){
-    throw std::runtime_error(e.what());
-  }
-  return false;
+bool CliHandler::get_io_term_active() const
+{
+    return io_term_active;
 }
 
-int CliHandler::get_bank_wise_sigma() const{
-  try{
-    return bank_wise_parms.at(1);
-  } catch (const std::exception& e){
-    throw std::runtime_error(e.what());
-  }
-  return false;
+
+const std::string& CliHandler::get_mem_spec_path() const
+{
+    return mem_spec_path;
 }
 
-bool CliHandler::get_pasr_active() const{
-  return pasr_active;
+const std::string& CliHandler::get_cmd_trace_path() const
+{
+    return cmd_trace_path;
 }
 
-int CliHandler::get_pasr_mode() const{
-  return pasr_mode;
+void CliHandler::loadMemSpec(const std::string &memspecUri)
+{
+    json doc = traceparser.parseJSON(memspecUri);
+    json jMemSpec = doc["memspec"];
+
+    std::string memoryType = jMemSpec["memoryType"];
+
+    if (memoryType == "DDR3") {
+        MemSpecDDR3 memSpecDDR3(jMemSpec,(get_writeToConsole() | get_writeToFile()),
```

Comment:
- Why are get_writeToConsole etc relevant for the memspec. This information should go to the library instead.
- We shoud think on the following: 
    - Lets keep the memspec as in DRAMSys as a pure data class
    - Introduce a configuration loader that parses the memspec json and fills the memspec data objects (code of DRAMSys could be reused here)